cluster/src/dunit/scala/io/snappydata/cluster/SecureQueryRoutingDUnitTest.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.cluster
+
+import java.sql.{Connection, DriverManager, ResultSet}
+
+import com.pivotal.gemfirexd.{Attribute, TestUtil}
+import com.pivotal.gemfirexd.security.LdapTestServer
+import io.snappydata.test.dunit.AvailablePortHelper
+
+import org.apache.spark.Logging
+import org.apache.spark.sql.collection.Utils
+
+class SecureQueryRoutingDUnitTest(val s: String)
+    extends ClusterManagerTestBase(s) with Logging {
+
+  override def setUp(): Unit = {
+    val ldapServer = LdapTestServer.getInstance(getClass.getResource("/auth.ldif").getPath)
+    if (!ldapServer.isServerStarted) {
+      ldapServer.startServer()
+    }
+    setSecurityProps(ldapServer)
+    super.setUp()
+  }
+
```

Comment:
Unless you are changing network server properties, you can "override val stopNetServersInTearDown = false" and explicitly stop all network servers in beforeClass/afterClass. This will also help check somewhat longer running VMs with different tests (and their cleanup) as more are added to this class.